src/bid/bid.rs

```diff
@@ -60,6 +59,81 @@ impl Borrow<u64> for Bid {
     }
 }
 
+impl PartialEq for Bid {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash().eq(&other.hash())
+    }
+}
+
+// This needs to be between braces since const fn calls passed as const_generics
+// params aren't perfectly supported yet.
+impl Serializable<{ Bid::serialized_size() }> for Bid {
```

Comment:
We should remove `serialized_size`, it was used before `Serializable` and the `SIZE` constant. We should have here directly the size or the count. For example:
```rust
impl Serializable<{PoseidonCipher::SIZE + 64 + 32 * 3 + 8 * 3}> for Bid {
```
And, if the types are also impl `Serializable`, you could actually have:
```rust
impl Serializable<{PoseidonCipher::SIZE + StealthAddress::SIZE + BlsScalar::SIZE * 3 + 8 * 3}> for Bid {
```
Basically the whole point is having one entry point when we impl the trait that tells exactly the size of the struct immediately, and what are the types involved.

For that reason (and also for error related ones) I was also thinking to implement `Serializable` for `u64` since we're using often, that would normalize a lot I guess. What do you think?

Reply:
This is really ugly TBH. But at the same time I agree simplifies and canonicallizes our APIs. Will go in that direction then. 

> For that reason (and also for error related ones) I was also thinking to implement Serializable for u64 since we're using often, that would normalize a lot I guess. What do you think?

 dusk-network/dusk-bytes#12